integrations/pytorch/tutorials/woodfisher_pruning.md

```diff
@@ -0,0 +1,126 @@
+<!--
+Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# WoodFisher Pruning with SparseML
+
+This tutorial shows how to use the WoodFisher pruning algorithm in a SparseML PyTorch
+pruning recipe. The WoodFisher algorithm was introduced in the paper 
+[WoodFisher: Efficient Second-Order Approximation for Neural Network Compression](https://arxiv.org/pdf/2004.14340.pdf)
+by Sidak Pal Singh and Dan Alistarh.
+
+## Overview
+The WoodFisher algorithm introduces an efficient process for approximating accurate
+second-order information about a model's weights that can be used to solve for optimal
+weights to prune.  The method uses the weights' first-order information
+(gradients) to approximate the Inverse-Hessian matrix that is used to score weights
+for pruning.
+
+This method was shown to achieve SoTA results for pruning tasks on
+image classification models such as ResNet and MobileNet. Detailed information of the
+methods and results can be found in the [paper](https://arxiv.org/pdf/2004.14340.pdf).
+
+WoodFisher pruning can be enabled by a single modifier in a SparseML recipe. This tutorial
+will demonstrate how to adapt an existing SparseML pruning recipe to do this.
+
+## WoodFisher Pruning Recipes
+To use the WoodFisher algorithm in a SparseML pruning recipe, the `MFACPruningModifier`
+must be used in place of the existing pruning modifier(s).  the `MFACPruningModifier`
+contains the same parameters as SparseML's magnitude pruning modifiers for controlling
+epoch range, sparsity levels, and schedule, so these values may remain
+unchaged when adjusting a recipe.
+
+The `MFACPruningModifier` implements a global pruning scheme however, so all
+parameters to be pruned should be listed under the `params` argument. If the previous
+recipe has multiple pruning modifiers, all of their prameters should be merged into a
+single `MFACPruningModifier`.
+
+WoodFisher specific hyperparameters can be specified under the `mfac_options` parameter.
+Values should be listed as a YAML dictionary.  Full detials about these hyperparameters
+are listed in the next section of this tutorial.
+
+### Example Recipe Section
+
+The following is an example `MFACPruningModifier` to be used in place of other
+PruningModifiers in a recipe:
```

Comment:
PruningModifiers is one string here right?

Reply:
changing to two lower case strings for consistency. It may be one in the future